src/api.rs

```diff
@@ -14,6 +17,42 @@ pub enum DriverType {
     Xen,
 }
 
+/// Supports passing initialization parameters to the driver
+///
+/// Some drivers can support optional extra initialization parameters.
+///
+/// This is required to initialize the KVM driver, which needs a `domain_name` and
+/// a `kvm_socket` parameters.
+///
+/// This is equivalent to LibVMI's `vmi_init_data_type_t`
+#[repr(C)]
+#[derive(Debug)]
+pub enum DriverInitParam {
+    KVMiSocket(String),
+}
+
+impl TryFrom<DriverInitParamFFI> for DriverInitParam {
+    type Error = IntoStringError;
+
+    fn try_from(param: DriverInitParamFFI) -> Result<Self, Self::Error> {
+        Ok(match param {
+            DriverInitParamFFI::KVMiSocket(cstr_socket) => DriverInitParam::KVMiSocket(
+                unsafe { CStr::from_ptr(cstr_socket) }
+                    .to_owned()
+                    .into_string()?,
+            ),
+        })
+    }
+}
```

Comment:
I kinda gave weird advice to be honest. What I said about the `From` trait does not apply to the `TryFrom` trait as there is no blanket implementation for `TryInto`. So conversely to `From` you do not get `TryInto` for free. Therefore I would suggest:
```suggestion
impl TryInto<DriverInitParam> for DriverInitParamFFI {
    type Error = IntoStringError;

    fn try_into(self) -> Result<DriverInitParam, Self::Error> {
        Ok(match self {
            DriverInitParamFFI::KVMiSocket(cstr_socket) => DriverInitParam::KVMiSocket(
                unsafe { CStr::from_ptr(cstr_socket) }
                    .to_owned()
                    .into_string()?,
            ),
        })
    }
}
```
And I would also move it to the capi.